packages/@lwc/integration-karma/test/component/LightningElement/index.spec.js

```diff
@@ -81,12 +81,19 @@ it("[W-6981076] shouldn't throw when a component with an invalid child in unmoun
     expect(() => document.body.removeChild(elm)).not.toThrow();
 });
 
-// TODO [W-17769475]: Restore this test when we can reliably detect Locker enabled
-xit('should fail when the constructor returns something other than an instance of itself', () => {
+it('should fail when the constructor returns something other than LightningElement when ENABLE_LIGHTNING_CONSTRUCTOR_CHECK is true', () => {
+    setFeatureFlagForTest('ENABLE_LIGHTNING_CONSTRUCTOR_CHECK', true);
     expect(() => {
         createElement('x-returning-bad', { is: ReturningBad });
     }).toThrowError(
         TypeError,
         'Invalid component constructor, the class should extend LightningElement.'
     );
+    setFeatureFlagForTest('ENABLE_LIGHTNING_CONSTRUCTOR_CHECK', false);
+});
+
+it('should succeed when the constructor returns something other than LightningElement when ENABLE_LIGHTNING_CONSTRUCTOR_CHECK is falsy', () => {
+    expect(() => {
+        createElement('x-returning-bad', { is: ReturningBad });
+    }).not.toThrow();
```

Comment:
```suggestion
    expect(createElement('x-returning-bad', { is: ReturningBad })).toBeInstanceOf(HTMLElement);
```
dunno if this is the exact assertion, but we should do a positive check rather than a negative one

Reply:
You sure? There are plenty of other `.not.toThrow` cases in this .spec file so it's consistent at least in the file context.